crates/sui-core/src/db_checkpoint_handler.rs

```diff
@@ -160,6 +173,45 @@ impl DBCheckpointHandler {
         }
         Ok(())
     }
+    async fn run_db_checkpoint_cleanup_loop(
+        self: Arc<Self>,
+        mut recv: tokio::sync::broadcast::Receiver<()>,
+    ) -> Result<()> {
+        let mut interval = tokio::time::interval(self.interval);
+        info!("DB checkpoint upload disabled. DB checkpoint cleanup loop started");
+        loop {
+            tokio::select! {
+            _now = interval.tick() => {
+                let local_checkpoints_by_epoch =
+                    find_all_dirs_with_epoch_prefix(&self.input_object_store, None).await?;
+                let mut dirs: Vec<_> = local_checkpoints_by_epoch.iter().collect();
+                dirs.sort_by_key(|(epoch_num, _path)| *epoch_num);
+                for (epoch, db_path) in dirs {
+                    // If db checkpoint marked as completed, skip
+                    let local_db_path = path_to_filesystem(self.input_root_path.clone(), db_path)?;
+                    let upload_completed_path = local_db_path.join(UPLOAD_COMPLETED_MARKER);
+                    if upload_completed_path.exists() {
+                        continue;
+                    }
+                    if self.prune_and_compact_before_upload {
+                        // Invoke pruning and compaction on the db checkpoint
+                        self.prune_and_compact(local_db_path, *epoch).await?;
+                    }
```

Comment:
Yes but I assumed it would be faster to scan a pruned and compacted db in order to make the formal snapshot. Specifically, the assumption is that pruning and compacting is faster than the additional time that would be required to scan a non compacted db. Does this assumption not hold?